src/main/java/org/opensearch/security/authtoken/jwt/JwtVendor.java

```diff
@@ -123,6 +127,38 @@ static JsonWebKey createJwkFromSettings(Settings settings) throws Exception {
         }
     }
 
+    private static JsonWebKey getSigningJwk(String algorithm, Settings settings) throws Exception {
+        JsonWebKey jwk = new JsonWebKey();
+        jwk.setPublicKeyUse(PublicKeyUse.SIGN);
+
+        if(AlgorithmUtils.isOctet(algorithm) ) {
+            jwk.setKeyType(KeyType.OCTET);
+            String signingKey = Optional.ofNullable(settings.get("signing_key")).orElseThrow(() -> new Exception("Settings for signing key is missing. Please specify at least the option signing_key with a shared secret."));
+            jwk.setProperty("k", signingKey);
+            jwk.setAlgorithm(algorithm);
+        } else if(AlgorithmUtils.isEcDsaSign(algorithm)) {
+            jwk.setKeyType(KeyType.EC);
+            jwk.setAlgorithm(algorithm);
+            jwk.setProperty(EC_CURVE, calcCurve(algorithm));
+            jwk.setProperty(EC_PRIVATE_KEY, Optional.ofNullable(settings.get("ec_private")).orElseThrow(() -> new Exception("Settings for EC private key is missing. Please specify ec_private with required x and y coordinates")));
+            jwk.setProperty(EC_X_COORDINATE, Optional.ofNullable(settings.get("ec_x")).orElseThrow(() -> new Exception("Settings for EC x coordinate is missing")));
+            jwk.setProperty(EC_Y_COORDINATE, Optional.ofNullable(settings.get("ec_y")).orElseThrow(() -> new Exception("Settings for EC y coordinate is missing")));
+        } else if(AlgorithmUtils.isRsaSign(algorithm)) {
+            //TODO decide on format of rsa key, should it be represented as modulus, first prime factor, second prime factor etc. as in this example: https://datatracker.ietf.org/doc/html/rfc7517#appendix-A or other format like PEM
+            jwk.setKeyType(KeyType.RSA);
+        } else {
+            throw new Exception("Signing algorithm not recognized: " + algorithm);
+        }
+        return jwk;
+    }
+
+    private static String calcCurve(String algorithm) throws Exception {
+        if(algorithm.equals(ES_SHA_256_ALGO)) return EC_CURVE_P256;
+        if(algorithm.equals(ES_SHA_384_ALGO)) return EC_CURVE_P384;
+        if(algorithm.equals(ES_SHA_512_ALGO)) return EC_CURVE_P521;
+        throw new Exception("Not recognized Elliptic Curve algorithm: " + algorithm);
```

Comment:
`Exception` is too broad. Could you please use `IllegalArgumentException` instead?

Reply:
agreed